stress_test/controller.py

```diff
@@ -0,0 +1,349 @@
+        else:
+            logging.info('[stop_controller] Controller already stopped.')
+
+
+    def wait_until_controller_listens(interval_ms, ssh_client=None):
+        """ Waits for controller to start listening on specified port.
+
+        :param interval_ms: milliseconds to wait (in milliseconds).
+        :param ssh_client : SSH client provided by paramiko to run the command
+        :raises Exception: If controller fails to start or if another process
+        listens on controllers port.
+        :rtype int
+        :type interval_ms: int
+        :type ssh_client: paramiko.SSHClient
+        """
+
+        timeout = time.time() + (float(interval_ms) / 1000)
+        while time.time() < timeout:
+            time.sleep(1)
+            gpid = util.process.getpid_listeningonport(self.port, ssh_client)
+            logging.info('Returned pid listening on port {0}: {1}'.
+                          format(self.port, gpid))
+            if gpid == 0:
+                raise Exception('Another controller seems to have started in the '
+                                'meantime. Exiting...')
+
+        raise Exception('Controller failed to start within a period of {0} '
+                        'minutes'.format(timeout))
+
+
+    def wait_until_controller_up_and_running(interval_ms, ssh_client=None):
+        """ Waits for controller status to become 1 (started).
+
+        :param interval_ms: milliseconds to wait (in milliseconds).
+        :param controller_status_handler: filepath to the controller status handler
+        :raises Exception: If controller fails to start or if another process
+        listens on controllers port.
+        :type interval_ms: int
+        :type ssh_client: paramiko.SSHClient
+        """
+
+        timeout = time.time() + (float(interval_ms) / 1000)
+        while time.time() < timeout:
+            time.sleep(1)
+            if check_controller_status(ssh_client) == '1':
+                return
+
+        raise Exception('Controller failed to start. '
+                        'Status check returned 0 after trying for {0} seconds.'.
+                        format(float(interval_ms) / 1000))
+
+
+
+class ODL(Controller):
+    def __init__(self,ctrl_base_dir,test_config):
+        Controller.__init__(self,test_config)
+
+        self.stat_period_ms = test_config['controller_statistics_period_ms']
+
+        if 'controller_flowmods_conf_handler' in test_config:
+            self.flowmods_conf_handler= ctrl_base_dir + test_config['controller_flowmods_conf_handler']
+
+        if 'controller_statistics_handler' in test_config:
+            self.statistics_handler = ctrl_base_dir + test_config['controller_statistics_handler']
+
+        if 'controller_persistent_handler' in test_config:
+            self.persistent_handler = ctrl_base_dir + test_config['controller_persistent_handler']
+
+
+    def controller_pre_actions(self, controller_ssh_client, controller_cpus):
```

Comment:
OK, when called in a test case, this method does not help in readability. 

When we will refactor a test case, we will use the steps performed in this method directly, e.g.:
```
if ctrl.need_rebuild:
    ctrl.build(...)

ctrl.check_other_active()
ctrl.generate_xml_files()
...
```
and so on